app/controlplane/internal/service/attestation.go

```diff
@@ -361,6 +361,31 @@ func (s *AttestationService) GetUploadCreds(ctx context.Context, req *cpAPI.Atte
 	return &cpAPI.AttestationServiceGetUploadCredsResponse{Result: resp}, nil
 }
 
+// completeRobotAccount auxiliary method to complete a RobotAccount instance. This is needed since when coming from
+// an API Token request, the only information we have set is the workflow name not the ID
+func (s *AttestationService) completeRobotAccount(ctx context.Context, workflowName string) (*usercontext.RobotAccount, error) {
```

Comment:
I find this method a little bit unnecessary now that it's used in one place only. It feels like too much to mutate an object to just get the workflowID. It could be simpler to add some of that code to the actual caller, and focus on that, get the workflowID not to mutate the robot account object. Does it make sense?

Reply:
Yes, it makes total sense!